### DISCUSSION TITLE
Page Management - Page Import/Export/Templates

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#3441] [Ideas]
Wondering if Page Management import and export will get on the road map?

It would be nice to save a page as a template as well.

It would need to handle events like when a module or framework feature no longer exists to skip trying to create the page with it.

An enhancement issue for on the list of things to do should be created if so.

**Comment by sbwalker** [comment]
No, this will not be added to the roadmap. There are already multiple options available for automating the creation of pages within a site:

https://www.oqtane.org/blog/!/62/routable-modules
https://www.oqtane.org/blog/!/38/site-migrations-deployment

**Comment by sbwalker** [comment]
This is a good example of an area where Oqtane does not intend to emulate DNN. The serialization of sites/pages/modules in DNN was hugely problematic to develop/test/maintain over time. As a result Oqtane is taking a different approach in this area. Since Oqtane's primary audience is software developers and not content managers, it is providing extensibility points, APIs, and interfaces to allow developers to automate the creation of sites/pages/modules.

**Comment by thabaum** [comment]
I have pretty much forgotten about DNN at this point as it is a thing of the past for me.  I studied Orchard Core as well, dang that took a long time to build and wow does it have class libraries!   Anyhow those and WordPress sites along with all the other CMS systems out there paid or not I have 0 interest in at this point.

My interest lies mainly in creating an app that can be 100% mobile and have a website work from it, which is why I chose Blazor to follow and study how I am going to implement an long term project that makes the most practical solution for building and maintaining it.  Oqtane is really the only option that is close to a CMS I am interested in possibly moving forward with.  .NET 8 changes will play a big part in my decision making as I still have pure JavaScript/Node.js for that same project.  I am trying not to throw away my C# knowledge but I cant say I am not thinking about it.

That blog for site migrations is great stuff, I was thinking that Oqtane could have a site replicator or something that would copy the site settings/features required to make a template there could be an option so we can more easily have more than just the Oqtane Default Template and the Blank Template without having to program them.

Thank you for the clarification and information that relates I will keep note of this and test a few scenarios out over the next week using these two solutions provided!
